### DISCUSSION TITLE
Mixed element types?

### DISCUSSION THREAD

**Discussion by joe61vette** [idaholab/moose#17478] [Q&A Meshing] [answered]
I am using Cubit and for one region I would like to use "tetmesh".  When I do, the result is a mixture of "tets" and "pyramids".  This region would be one block for which I would normally set an element type.  But..

Is it possibly to run a problem with a MOOSE-based code that has such a mix of element types?  If so, how should I set the element type?

Thanks,
  Joe Kelly

**Comment by WilkAndy** [comment]
You can certainly use MOOSE to run a model containing a mix of element types, and you shouldn't need to set the element type manually - MOOSE will just handle everything automatically.  There are only a few examples of this in the tests, for instance, many of the tests using `AnnularMeshGenerator` that creates QUAD4 and TRI3 elements in a single mesh.  However,  because of the constraints from the Exodus mesh definition, you're going to have to put your TETs into one block (also called "subdomain" in parts of MOOSE) and PYRAMIDs into another block.  By the way, this gives you the freedom of prescribing different physics to the TETs than the PYRAMIDs, if you needed to, although it doesn't sound like you actually want to do that.

**Reply by joe61vette** [reply]
Thanks Andy.

Unfortunately, there is no easy way to separate the pyramids from the tets.  The pyramids get automatically generated to bridge from a neighboring region that uses quads. Maybe it is possible to convert the pyramids to tets?  I'll look at doing that.

 Joe

**Reply by WilkAndy** [reply]
So this has become a CUBIT question.  I'm not a CUBIT expert - is there some way to "select all PYRAMIDs within a region and add them to a block"?

**Reply by joe61vette** [reply]
Thanks again Andy.  I found a way to mesh this to avoid using tetmesh.  So, for now at least, my problem is solved.  May have to revisit this again later as pyramids provide a great transition from quads.  As I have a lot of long relatively thin regions, a map mesh that is swept is generally the best way to go. Unfortunately, there are sometimes thin wedge shaped adjoining regions for which tets are the best.  Pyramids provide the transition but with no clearly defined boundary.

  Joe

**Comment by joe61vette** [comment]
Andy, thanks to your suggestion about finding a command to group all pyramids in a block, I believe that I have found a solution for this.  Namely:

Block 1 add volume 1                    # puts all elements of volume 1 in block 1
Block 1 remove pyramid all           # removes all pyramid elements from block 1

Block 2 add volume 1                   # puts all elements of volume 1 in block 2
Block 2 remove tet all                  # removes all tet elements from block 2

I think that will solve my problem.  You know Cubit has a lot of capability but sometimes it is difficult for a casual user to find it.

  Joe

**Comment by joe61vette** [comment, marked as answer]
Hello Again:

Even better, Cubit can actually do this for you automatically.  Just use:

set block mixed element output offset

It then adds an "offset" to the block ID.  One can specify the value of the offset (eg, 1000) for each element type.

 Joe

**Reply by permcody** [reply]
Yes, I think I remember doing this years ago. Thanks for the reminder Joe!
